src/include/storage/checkpoints/checkpoint_thread.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
```

Comment:
I'm curious if there is a thread (or maybe it is called a task?) that is used by the general recovery/log manager that could be used for checkpoints, since it seems that checkpointing would be a fluid part of the recovery process, and  not in temporal conflict with logs being written to disk.

Reply:
We have changed the CheckpointThread class to use an existing thread pool